rmc/utils/constraint.py

```diff
@@ -955,214 +955,199 @@ def get_rescue_transcripts_and_create_no_breaks_ht(overwrite: bool) -> None:
+    Get table of final RMC sections in a given pathway (initial or rescue) after all break searches are complete.
 
-    .. note::
-        - This function will only get unique transcripts for transcripts with one or one + additional breaks.
-        - This will not work for transcripts with two simultaneous breaks.
-        - Assumes input Table is annotated with list containing booleans for whether that locus is a breakpoint
-        (`break_list`).
+    :param round_nums: List of round numbers to merge results across.
+    :param is_rescue: Whether to operate on search in rescue pathway.
 
-    :param ht: Input Table.
-    :param max_n_breaks: Largest number of breaks.
-    :return: Dictionary with break number (key) and set of transcripts unique to that break number or empty SetExpression (value).
+    :return: Table of final RMC sections. Schema:
+        ----------------------------------------
+        Row fields:
+            'section_obs': int64
+            'section_exp': float64
+            'section_oe': float64
+            'section_chisq': float64
+            'search_type': str
+            'transcript': str
+            'interval': interval<locus<GRCh37>>
+        ----------------------------------------
+        Key: ['transcript', 'interval']
+        ----------------------------------------
     """
-    transcripts_per_break = {}
-    # Use `break_list` annotation (list of booleans for whether a row is a breakpoint)
-    # to filter one/additional breaks ht to rows that are significant breakpoints ONLY
-    ht = ht.filter(ht.break_list.any(lambda x: x))
-
-    # Group HT (filtered to breakpoint positions only) by transcript and
-    # count the number of breakpoints associated with each transcript
-    group_ht = ht.group_by("transcript").aggregate(n_breaks=hl.agg.count())
-
-    # Checkpoint to force hail to finish this group by computation
-    group_ht = group_ht.checkpoint(
-        f"{TEMP_PATH}/breaks_per_transcript.ht", overwrite=True
+    if len(round_nums) < 2:
+        raise DataException(
+            "At least two rounds of break search are needed if evidence of RMC is found, please double-check!"
+        )
+
+    logger.info(
+        "Collecting and merging no-break HTs from each search round starting at round 2..."
+    )
+    hts = []
+    for search_num in round_nums[1:]:
+        # For each search round:
+        # Get locus-level merged no-break table (no breaks in both single and simultaneous searches)
+        ht = merge_round_no_break_ht(is_rescue=False, search_num=search_num)
+        # Group to section-level and retain section obs- and exp-related annotations
+        ht = ht.group_by("section").aggregate(
+            section_obs=hl.agg.take(ht.section_obs, 1)[0],
+            section_exp=hl.agg.take(ht.section_exp, 1)[0],
+            section_oe=hl.agg.take(ht.section_oe, 1)[0],
+        )
+        hts.append(ht)
+    rmc_ht = hts[0].union(*hts[1:])
```

Comment:
```suggestion
    rmc_ht = hts[0].union(*hts[1:])
    rescue = "rescue" if is_rescue else "initial"
    rmc_ht = rmc_ht.checkpoint(f"{TEMP_PATH_WITH_DEL}/{rescue}_search_union.ht", overwrite=True)
```
good to checkpoint here after the join (probably also fine to checkpoint after the chi square calculation/search_type annotation too though if you want to move this)